ghproxy/ghcache/coalesce.go

```diff
@@ -57,75 +57,79 @@ func (r *requestCoalescer) RoundTrip(req *http.Request) (*http.Response, error)
 	}
 
 	var cacheMode = ModeError
-	defer func() {
-		cacheCounter.WithLabelValues(cacheMode).Inc()
-	}()
-
-	key := req.URL.String()
-	r.Lock()
-	waiter, ok := r.keys[key]
-	if ok {
-		// Earlier request in flight. Wait for it's response.
-		if req.Body != nil {
-			defer req.Body.Close() // Since we won't pass the request we must close it.
+	resp, err := func() (*http.Response, error) {
```

Comment:
This diff looks worse than it is. I just moved most of the logic into this anonymous func to get rid of the named return values and defer magic.